nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/NotificationEventListener.java

```diff
@@ -237,9 +239,9 @@ protected void sendNotificationSignalForUser(Notification notification, String s
         log.debug("Producing notification message.");
 
         Map<String, Serializable> eventInfo = ctx.getProperties();
-        DocumentModel doc = ctx.getSourceDocument();
+        DocumentModel doc = (DocumentModel) ctx.getProperty(TOP_LEVEL_DOCUMENT);
         String author = ctx.getPrincipal().getName();
-        Calendar created = (Calendar) ctx.getSourceDocument().getPropertyValue("dc:created");
+        Calendar created = (Calendar) doc.getPropertyValue("dc:created");
```

Comment:
i think we shouldn't rely on `TOP_LEVEL_DOCUMENT ` it is not the responsibility of `NotificationEventListener` to get the `TOP_LEVEL_DOCUMENT ` this information is only related to `Comment` and `Annotation`. if you want others informations then we should compute them outer `NotificationEventListener` like the `dc:created` ....and then you should keep `Calendar) ctx.getSourceDocument().getPropertyValue("dc:created")` for backward compatibility.